concent_api/core/views.py

```diff
@@ -529,6 +541,99 @@ def handle_send_force_subtask_results_results_response(request, client_message):
+        return message.concents.ForcePaymentRejected(  # Provider receives ForcePaymentRejected from Concent.
+            reason = message.concents.ForcePaymentRejected.REASON.TimestampError
+        )
+
+    list_of_forced_payments = _get_forced_payments(T0) # Concent gets list of list of forced payments from payment API where T0 <= payment_ts + PAYMENT_DUE_TIME + PAYMENT_GRACE_PERIOD.
+
+    sum_of_payments_from_subtask_results_accepted   = sum_payments(client_message.subtask_results_accepted_list)
+    sum_of_payments_from_list_of_transactions       = sum_payments(list_of_transactions)
+    sum_of_payments_from_list_of_force_payments     = sum_payments(list_of_forced_payments)
+
+    V = sum_paymets_from_subtask_results_accepted - sum_of_payments_from_list_of_transactions - sum_of_payments_from_list_of_force_payments
+
+    T2 = min(subtask_results_accepted.payment_ts for subtask_results_accepted in client_message.subtask_results_accepted_list) # Concent defines time T2 (end time) equal to youngest payment_ts from passed SubtaskResultAccepted messages from subtask_results_accepted_list.
+
+    if V <= 0:
+        return message.concents.ForcePaymentRejected(
+            reason = message.concents.ForcePaymentRejected.REASON.NoUnsettledTasksFound
+        )
+    elif V > 0:
+        response = _make_payment_to_provider(V, requestor_ethereum_public_key, client_public_key)
+        provider_force_payment_commited = message.concents.ForcePaymentCommitted(
+            payment_ts              = current_time + 60,
+            task_owner_key          = requestor_ethereum_public_key,
+            provider_eth_account    = client_public_key,
+            amount_paid             = 10.99,
+            amount_pending          = 0.01,
+            recipient_type          = message.concents.ForcePaymentCommitted.Actor.Provider,
+        )
+
+        store_message_and_message_status(
+            provider_force_payment_commited.TYPE,
+            None,
+            provider_force_payment_commited.serialize(),
+            provider_public_key     = client_public_key,
+            requestor_public_key    = client_message.subtask_results_accepted_list[0].task_to_compute.compute_task_def['requestor_public_key'],
+        )
+
+        return provider_force_payment_commited
+
+
+def _get_list_of_transactions(_T0):
+    '''
+    Function which return list of transactions from payment API
+    where timestamp >= T0
+    '''
+    pass
+
+def _get_forced_payments(_T0):
+    '''
+    Function which return list of forced paysments from payment API
+    where t0 <= payment_ts + PAYMENT_DUE_TIME + PAYMENT_GRACE_PERIOD
+    '''
+    pass
+
+def _make_payment_to_provider(_V, requestor_ethereum_public_key, provider_ethereum_public_key):
```

Comment:
Those functions names looks good, at the end they should be in separate module ie. `payments\base.py`, and each backend in separate file ie. `payments\mock.py`, `payments\sci.py` etc.

Reply:
At the end of sprint they should be seperated or when we get working API?